driver-core/src/main/java/com/datastax/driver/core/StatementFormatter.java

```diff
@@ -0,0 +1,1108 @@
+                }
+            }
+        }
+
+        protected BatchStatement.Type getBatchType(BatchStatement statement) {
+            return statement.batchType;
+        }
+
+        protected List<List<ByteBuffer>> getValues(BatchStatement statement, StatementWriter out) {
+            return statement.getIdAndValues(out.getProtocolVersion(), out.getCodecRegistry()).values;
+        }
+
+    }
+
+    public static class StatementWrapperPrinter implements StatementPrinter<StatementWrapper> {
+
+        @Override
+        public Class<StatementWrapper> getSupportedStatementClass() {
+            return StatementWrapper.class;
+        }
+
+        @Override
+        public void print(StatementWrapper statement, StatementWriter out, StatementFormatVerbosity verbosity) {
+            Statement wrappedStatement = statement.getWrappedStatement();
+            StatementPrinter<? super Statement> printer = out.getPrinterRegistry().findPrinter(wrappedStatement);
+            printer.print(wrappedStatement, out, verbosity);
+        }
+
+    }
+
+    public static class SchemaStatementPrinter extends AbstractRegularStatementPrinter<SchemaStatement> {
+
+        public SchemaStatementPrinter() {
+            super(SchemaStatement.class);
+        }
+
+    }
+
+    public static class RegularStatementPrinter extends AbstractRegularStatementPrinter<RegularStatement> {
+
+        public RegularStatementPrinter() {
+            super(RegularStatement.class);
+        }
+
+    }
+
+    public static class DefaultStatementPrinter extends StatementPrinterBase<Statement> {
+
+        public DefaultStatementPrinter() {
+            super(Statement.class);
+        }
+
+    }
+
+    /**
+     * Helper class to build {@link StatementFormatter} instances with a fluent API.
+     */
+    public static class Builder {
+
+        public static final int DEFAULT_MAX_QUERY_STRING_LENGTH = 500;
+        public static final int DEFAULT_MAX_BOUND_VALUE_LENGTH = 50;
+        public static final int DEFAULT_MAX_BOUND_VALUES = 10;
+        public static final int DEFAULT_MAX_INNER_STATEMENTS = 5;
+        public static final int DEFAULT_MAX_OUTGOING_PAYLOAD_ENTRIES = 10;
+        public static final int DEFAULT_MAX_OUTGOING_PAYLOAD_VALUE_LENGTH = 50;
+
+        private int maxQueryStringLength = DEFAULT_MAX_QUERY_STRING_LENGTH;
+        private int maxBoundValueLength = DEFAULT_MAX_BOUND_VALUE_LENGTH;
+        private int maxBoundValues = DEFAULT_MAX_BOUND_VALUES;
+        private int maxInnerStatements = DEFAULT_MAX_INNER_STATEMENTS;
+        private int maxOutgoingPayloadEntries = DEFAULT_MAX_OUTGOING_PAYLOAD_ENTRIES;
+        private int maxOutgoingPayloadValueLength = DEFAULT_MAX_OUTGOING_PAYLOAD_VALUE_LENGTH;
+
+        private final List<StatementPrinter<?>> printers = new ArrayList<StatementPrinter<?>>();
+
+        private Builder() {
+        }
+
+        public Builder addStatementPrinter(StatementPrinter<?> printer) {
```

Comment:
Would be good to add javadoc here explaining why a user would register a custom statement printer.